### DISCUSSION TITLE
Access to data in custom metric function

### DISCUSSION THREAD

**Discussion by ooghry** [autogluon/autogluon#1834] [Q&A] [answered]
Hello,
I'm working on a personal algorithmic trading system. I have the `target` column that has three values: `buy, sell, and no_action` (rows are independent)
I also have a `profit` and `loss` column that shows If the system predicts correctly, how much will it gain and if it does wrong, how much will it lose.
Is it possible in autogluon to create custom metric function that access to those two column for calculation?

**Comment by Innixma** [comment, marked as answer]
> Hello, I'm working on a personal algorithmic trading system. I have the `target` column that has three values: `buy, sell, and no_action` (rows are independent) I also have a `profit` and `loss` column that shows If the system predicts correctly, how much will it gain and if it does wrong, how much will it lose. Is it possible in autogluon to create custom metric function that access to those two column for calculation?

We don't currently support this, and supporting it would probably be quite complicated. One option is to use this metric outside of AutoGluon to score what it predicts and choose the AutoGluon model that scores the best.

You could simplify your metric to make it work by instead of having `profit` and `loss` as separate things, instead average them to create a singular value (aka 10 profit, 6 loss = 2). You can then use this value, normalized across all rows, as your sample_weight, and then enable sample_weight and weight_evaluation via:

```
predictor = TabularPredictor(..., sample_weight=your_sample_weight_column_name, weight_evaluation=True).fit(...)
```

https://auto.gluon.ai/stable/api/autogluon.task.html#module-0

This isn't a perfect solution but may at least get you strong models.

**Reply by ooghry** [reply]
Thank you for your reply. I would consider your suggestion.

**Reply by ooghry** [reply]
Your explanation was almost clear, but I can not find any example of the usage of `sample_weight`. I'm not sure how the Autogluon considers values of `sample_weight`.
Creating a simple example of using `sample_weight` in docs would be nice.
